### ISSUE TITLE
Invalid use of aria-labelledby, target ID is missing

### ISSUE THREAD

**Issue by saprentice** [dita4publishers/dita4publishers#32]
In an EPUB build I'm getting the following ..

      <main role="main">
         <article role="article" aria-labelledby="ariaid-title1">
            <h1 class="title topictitle1"><span class="enumeration_chapter">Chapter 2.</span> Installation</h1>

There is no `id="ariaid-title1"`, so the `aria-labelledby="ariaid-title1"` is invalid. When I do a straight HTML5 transform, the ID is set on the H1 tag, but for some reason when I do an EPUB build, the ID is gone. This exists in all files, and thus creates a lot of noise with an epubcheck.

**Comment by drmacro** [comment]
Which version of the OT are you running with?

I just recently ran epubcheck against epubs generated with the 2.5.4 OT and got no messages, so I don't think this should occur with 2.5.4.

**Comment by drmacro** [comment]
Which version of the plugin are you using? You may need to get the latest from develop branch but i should have released the version that handles the aria attribute.

I use epubcheck to validate but I think Oxygen does the same validation.

**Comment by saprentice** [comment]
I'm using dita4publishers-epub-toolkit-plugins-1.0.0RC26-ot-2.4 .. but 
I'll grab the latest from the develop branch and see if that helps.

I'll be featuring this in my EPUB workshop at tcworld. LMK if you make 
any updates or update the release.

Enjoy DITA Europe!

Cheers,
...scott

**Comment by saprentice** [comment]
Hmm. Looks like I can't just "use" those files. Are there instructions 
for building this project?

Thanks,
...scott

**Comment by drmacro** [comment]
Yes, building the plugins actually requires a bit of effort because of the way the submodules are organized. 

The instructions for setting up a dev environment are here: https://github.com/dita4publishers/dita4publishers/wiki (just realized there's no link from the readme, poor planning on my part).

Do you have a test doc set that demonstrates the problem?

I can test and make you a new build if I fix anything.

**Comment by drmacro** [comment]
The bad aria-labeledby value is a bug in the HTML5 transform:

```
 <xsl:template match="*[contains(@class, ' topic/topic ')]/*[contains(@class, ' topic/title ')]" mode="return-aria-label-id">
    <xsl:choose>
      <xsl:when test="@id">
        <xsl:sequence select="dita-ot:generate-id(parent::*/@id, @id)"/>
      </xsl:when>
      <xsl:otherwise>
          <xsl:text>ariaid-title</xsl:text>
        <xsl:number count="*[contains(@class, ' topic/title ')][parent::*[contains(@class,' topic/topic ')]]" level="any"/>
      </xsl:otherwise>
    </xsl:choose>
  </xsl:template>
```

This will produce a good value only if the *title* element has an ID.

I can patch this in the EPUB transform.

**Comment by drmacro** [comment]
However, I'm getting other epubcheck failures from the epub created from your sample, so I need to track those down as well.

**Comment by drmacro** [comment]
Actually, I misread the code. It should be setting an ID on the title element but that's not happening.

**Comment by drmacro** [comment]
Actually, with more tests, I can't reproduce the missing @id on h1 problem--the EPUB I'm generating is good with the current code on the develop branch.

**Comment by drmacro** [comment]
Should be corrected in release 1.0.0RC27
